zeppelin-cluster/yarn/src/main/java/org/apache/zeppelin/cluster/yarn/RemoteInterpreterYarnProcess.java

```diff
@@ -0,0 +1,512 @@
+    this.interpreterDir = interpreterDir;
+    this.configuration = configuration;
+    this.name = name;
+    this.group = group;
+    this.env = env;
+    this.properties = properties;
+
+    this.waitingInitialized = new CountDownLatch(1);
+    this.interpreterLibPaths = Lists.newArrayList(
+        Paths.get(homeDir, "zeppelin-interpreter", "target"),
+        Paths.get(homeDir, "zeppelin-interpreter", "target", "lib"),
+        Paths.get(homeDir, "zeppelin-cluster", "common", "target"),
+        Paths.get(homeDir, "zeppelin-cluster", "common", "target", "lib"),
+        Paths.get(homeDir, "zeppelin-cluster", "yarn", "target"),
+        Paths.get(homeDir, "zeppelin-cluster", "yarn", "target", "lib"),
+        Paths.get(homeDir, "conf", "yarn", "log4j.properties"),
+        Paths.get(homeDir, "lib", "interpreter"),
+        Paths.get(homeDir, "lib", "cluster", "common"),
+        Paths.get(homeDir, "lib", "cluster", "yarn"));
+  }
+
+  @Override
+  public String getHost() {
+    return host;
+  }
+
+  private void setHost(String host) {
+    this.host = host;
+  }
+
+  @Override
+  public int getPort() {
+    return port;
+  }
+
+  private void setPort(int port) {
+    this.port = port;
+  }
+
+  @Override
+  public void start(String userName, Boolean isUserImpersonate) {
+    if (isUserImpersonate) {
+
+    }
+    try {
+      YarnClientApplication app = yarnClient.createApplication();
+      ApplicationSubmissionContext appContext = app.getApplicationSubmissionContext();
+
+      // put this info idApplicationIdMap
+      ApplicationId applicationId = appContext.getApplicationId();
+
+      appContext.setKeepContainersAcrossApplicationAttempts(false);
+      appContext.setApplicationName(name);
+
+      Map<String, String> env = Maps.newHashMap();
+
+      ArrayList<String> classpathStrings = Lists.newArrayList(configuration
+          .getStrings(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
+              YarnConfiguration.DEFAULT_YARN_CROSS_PLATFORM_APPLICATION_CLASSPATH));
+      classpathStrings.add(0, "./*");
+      classpathStrings.add(0, ApplicationConstants.Environment.CLASSPATH.$$());
+      classpathStrings.add("./log4j.properties");
+      classpathStrings.add(System.getenv("HADOOP_CONF_DIR"));
+      classpathStrings.add("`hadoop classpath 2> /dev/null`");
+
+      String classpathEnv =
+          Joiner.on(ApplicationConstants.CLASS_PATH_SEPARATOR).join(classpathStrings);
+
+      logger.debug("classpath: {}", classpathEnv);
+
+      env.putAll(this.env);
+      env.put("CLASSPATH", classpathEnv);
+
+      if (isHadoopConfSet()) {
+        env.put("HADOOP_CONF_DIR", System.getenv("HADOOP_CONF_DIR"));
+      }
+
+      Map<String, LocalResource> localResources = new HashMap<>();
+
+      FileSystem fileSystem = FileSystem.get(configuration);
+
+      Path interpreterDir = getInterpreterRelativePath();
+      List<Path> interpreterPaths = getPathsFromDirPath(interpreterDir);
+
+      if (isSparkInterpreter()) {
```

Comment:
@zjffdu Yes, we will support yarn cluster mode using spark-submit. But I thought we need to support running spark like other interpreters. After we finish to implement yarn-cluster mode feature for Spark Interpreter, it will be useless and then we will remove this kind of tricky code, too.